lib/inputs/rpm/static.ts

```diff
@@ -0,0 +1,88 @@
+import { unlink, writeFile } from "fs";
+import { EOL, tmpdir } from "os";
+import { resolve as resolvePath } from "path";
+import { getContent } from "../../extractor";
+import { ExtractAction, ExtractedLayers } from "../../extractor/types";
+import { streamToBuffer } from "../../stream-utils";
+import { CmdOutput, execute } from "../../sub-process";
+import { RpmDbPackage } from "./types";
+
+export const getRpmDbFileContentAction: ExtractAction = {
+  actionName: "rpm-db",
+  fileNamePattern: "/var/lib/rpm/Packages",
+  callback: streamToBuffer,
+};
+
+export async function getRpmDbFileContent(
+  extractedLayers: ExtractedLayers,
+): Promise<string> {
+  const apkDb = getContent(extractedLayers, getRpmDbFileContentAction);
+  if (!apkDb) {
+    return "";
+  }
+
+  const filePath = generateTempFileName();
+  await writeToFile(filePath, apkDb as Buffer);
+
+  try {
+    const cmdOutput = await execute("rpmdb", ["-f", filePath]).catch(
+      handleError,
+    );
+    return formatRpmDbOutput(cmdOutput);
+  } finally {
+    await removeFile(filePath);
+  }
+}
+
+function handleError(error): CmdOutput | never {
+  const stderr = error.stderr;
+  if (typeof stderr === "string" && stderr.indexOf("not found") >= 0) {
+    return { stdout: "", stderr: "" };
+  }
+  throw error;
+}
+
+function generateTempFileName(): string {
+  const tmpPath = tmpdir();
+  const randomFileName = Math.random().toString();
+
+  return resolvePath(tmpPath, randomFileName);
+}
+
+function writeToFile(filePath: string, apkDb: Buffer): Promise<string> {
```

Comment:
shouldn't all of those functions be async? I know it doesn't matter but just for consistency

Reply:
This one shouldn't be async because we directly return the promise, we don't have to await it because the caller of this function decides whether to await

Reply:
They aren't the same I think. To make the function async I have to rewrite the code as such:
```js
async function(): Promise<string> {
  const result: string = await new Promise((resolve) => {
    ... etc. etc.
    resolve(string_result);
  }
  return result;
}
```

Whereas right now it is...
```js
function(): Promise<string> {
  return new Promise((resolve) => {
    ... etc. etc.
  }
}

somewhere else:
const result: string = await function();
```

Reply:
I think you can just add `async` to your function and it will work, both functions return `Promise<string>`